bindsnet/network/nodes.py

```diff
@@ -682,7 +683,8 @@ def forward(self, x: torch.Tensor) -> None:
         # Refractoriness, voltage reset, and adaptive thresholds.
         self.refrac_count.masked_fill_(self.s, self.refrac)
         self.v.masked_fill_(self.s, self.reset)
-        self.theta += self.theta_plus * self.s.float()
+        if self.network.learning:
```

Comment:
Using the condition in the forward method will slow the performance.
Its rather to check the condition one time and not twice on the same function.

We need to find other way to differentiate between training or testing, maybe two forward function? 
Or when setting `network.learning = False` some values will be zero or one depend on the usage.

Reply:
```%timeit if True: pass                                                   
5.54 ns ± 0.0233 ns per loop (mean ± std. dev. of 7 runs, 100000000 loops each)

Reply:
How about a function to switch between learning and testing? 
Then some parameters can be change between modes

Reply:
Maybe, but that will require some thought + more code changes. I think this is good for the time being, but we can schedule the train / test switching function for future work.